packages/NodeTypeResolver/NodeScopeAndMetadataDecorator.php

```diff
@@ -45,35 +34,16 @@ public function __construct(
      */
     public function decorateNodesFromFile(File $file, array $stmts): array
     {
-        $nodeTraverser = new NodeTraverser();
-        $nodeTraverser->addVisitor(new NameResolver(null, [
-            self::OPTION_PRESERVE_ORIGINAL_NAMES => true,
-            // required by PHPStan
-            self::OPTION_REPLACE_NODES => true,
-        ]));
-
-        /** @var Stmt[] $stmts */
-        $stmts = $nodeTraverser->traverse($stmts);
-
         $smartFileInfo = $file->getSmartFileInfo();
-
         $stmts = $this->phpStanNodeScopeResolver->processNodes($stmts, $smartFileInfo);
 
-        $nodeTraverserForPreservingName = new NodeTraverser();
-
-        $preservingNameResolver = new NameResolver(null, [
-            self::OPTION_PRESERVE_ORIGINAL_NAMES => true,
-            // this option would override old non-fqn-namespaced nodes otherwise, so it needs to be disabled
-            self::OPTION_REPLACE_NODES => false,
-        ]);
-
-        $nodeTraverserForPreservingName->addVisitor($preservingNameResolver);
-        $stmts = $nodeTraverserForPreservingName->traverse($stmts);
-
```

Comment:
Main point of this PR is to remove these 2 node name traversers. Why?

## Pros

* They already happen in PHPStan parser that we now use since PHPStan 1.0
    * https://github.com/phpstan/phpstan-src/blob/c471c7b050e0929daf432288770de673b394a983/src/Parser/RichParser.php    
    * https://github.com/phpstan/phpstan-src/blob/c471c7b050e0929daf432288770de673b394a983/src/Parser/SimpleParser.php

* Here are used to work around PHPStan changing names without keeping the originals, without those, Rector would print nodes in FQN forms everywhere

## Cons

It has downside in losing rules that rely deply on original node names without overriding, mainly with alias imports in union. Thus 2 rules had to removed:

* https://github.com/rectorphp/rector-src/pull/1160
* https://github.com/rectorphp/rector-src/pull/1157

Yet, I think it's worth it, as we cut 3 traverses to 1 and based on shared names with PHPStan traversing, making complexity lower in general.